pkg/obfuscator/obfuscator.go

```diff
@@ -6,6 +6,10 @@ type Obfuscator interface {
 	Path(string) string
 	// Contents takes string as input and return the obfuscated string
 	Contents(string) string
-	// Report returns a map of words and their replacements
+}
+
+type ReportingObfuscator interface {
+	Obfuscator
```

Comment:
If everything is a `ReportingObfuscator` why do we need 2 different interfaces?

Reply:
for logic that doesn't need reporting?

Reply:
Because the way I see it there are different types of obfuscators and they obfuscate stuff and also report what they did.